apps/mark-scan/backend/src/custom-paper-handler/state_machine.ts

```diff
@@ -324,15 +369,87 @@ export function buildMachine(
           id: 'interpretScannedBallot',
           src: loadMetadataAndInterpretBallot,
           onDone: {
-            target: 'presenting_ballot',
+            target: 'transition_interpretation',
             actions: assign({
-              interpretation: (_, event) => {
-                return event.data;
-              },
+              interpretation: (_, event) => event.data,
             }),
           },
         },
       },
+      // Intermediate state to conditionally transition based on ballot interpretation
+      transition_interpretation: {
+        always: [
+          {
+            target: 'presenting_ballot',
+            cond: (context) =>
+              context.interpretation
+                ? context.interpretation[0].interpretation.type ===
+                  'InterpretedBmdPage'
+                : false,
+          },
+          {
+            target: 'blank_page_interpretation',
+            cond: (context) =>
+              context.interpretation
+                ? context.interpretation[0].interpretation.type === 'BlankPage'
+                : false,
```

Comment:
What happens if there is no interpretation? It seems like both conds would be false and we'd get stuck in this state. Seems like maybe we expect `context.interpretation` to always be defined, in which case I think an `assert` would be more clear

Reply:
I was thinking with the assert that you could remove the conditional check. So something like this:

```ts
assert(context.interpretation)
return context.interpretation[0].interpretation.type === 'BlankPage`
```
or even
```ts
assertDefined(context.interpretation)[0].interpretation.type === 'BlankPage'
```
Sorry I wasn't clear enough!

Reply:
Oops sorry your original comment was clear, just rushed the fix on my part. [700b335](https://github.com/votingworks/vxsuite/pull/3981/commits/700b3351b465b1f80fe509abb721c978ab0165ba) uses `assertDefined` and removes the conditional check.

It also adds an assertion on interpretation type in `entry`. It's redundant with the check in the `cond` transitions but
* it's valuable to throw on unexpected interpretation types. Without this assertion we'd get stuck in `transition_interpretation`
* Typescript isn't aware that `entry` is guaranteed to precede `cond` transitions so we need to re-assert with `assertDefined`